mri/reconstruct/fourier.py

```diff
@@ -209,3 +215,221 @@ def adj_op(self, x):
         """
         self.plan.f = x
         return np.copy(self.plan.adjoint()) / np.sqrt(self.plan.M)
+
+
+class Singleton:
```

Comment:
can you document that class? I feel that this a bit like a counter, am I right?

Reply:
Yes, it tells about the number of instances.

Reply:
The number of instances of what? Could you use `Counter` from `collections`?